### PR TITLE
Eco 1112/onboarding screen

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Create a new onboarding screen
#### Technical description:
- Refactored SplashAcitivty to OnboardingFragment
- Refactored KinEcosystemAcvitiy to use only fragments and be the main activity.
- Removed unused files
- Converted few files to Kotlin
#### Tests:
- Updated the relevant tests for onboarding screen

### CODE REVIEW

File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+interface SettingsDataSource {
+
+    fun setIsBackedUp(publicAddress: String, isBackedUp: Boolean)
+
+    fun isBackedUp(publicAddress: String): Boolean
+
+    fun setSawOnboarding(kinUserId: String)
+
+    fun isSawOnboarding(kinUserId: String) : Boolean
```
Comment (yosriz) [inline core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSource.kt:11]: something sound odd english wise, maybe `didSawOnboarding`?

File: sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        blockchainSource.addBalanceObserver(balanceObserver!!, false)
+    }
+
+    private fun isGreaterThenZero(value: Balance): Boolean {
+        return value.amount.compareTo(BigDecimal.ZERO) == 1
+    }
+
+    private fun updateMenuSettingsIcon() {
+        if (!StringUtil.isEmpty(publicAddress)) {
+            if (!settingsDataSource.isBackedUp(publicAddress!!)) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.kt]: can you remove the `!!` somehow?

**Review by yosriz — APPROVED** [review]
👍

File: sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.kt (hunk 2 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.kt]: fixed👍

**Pull request merged**